webserver/views/api/v1/test/test_core.py

```diff
@@ -274,14 +274,132 @@ def test_get_bulk_hl_absent_mbid(self, load_many_high_level):
+        same MBID. 
+
+        If metadata.audio_properties and metadata.version are not requested, 
+        they are still included. 
+        
+        Duplicate features and (MBID, offset) combinations are removed.""" 
+
+        recordings = "c5f4909e-1d7b-4f15-a6f6-1af376bc01c9;c5f4909e-1d7b-4f15-a6f6-1af376bc01c9:0;405a5ff4-7ee2-436b-95c1-90ce8a83b359:2;405a5ff4-7ee2-436b-95c1-90ce8a83b359:3"
+        features = "lowlevel.average_loudness;rhythm.onset_rate;lowlevel.average_loudness"
+
+        rec_c5 = {"recording": "c5f4909e-1d7b-4f15-a6f6-1af376bc01c9"}
+        rec_40_2 = {"recording": "405a5ff4-7ee2-436b-95c1-90ce8a83b359:2"}
+        rec_40_3 = {"recording": "405a5ff4-7ee2-436b-95c1-90ce8a83b359:3"}
+
+        expected_result = {
+            "c5f4909e-1d7b-4f15-a6f6-1af376bc01c9": {"0": rec_c5},
+            "405a5ff4-7ee2-436b-95c1-90ce8a83b359": {"2": rec_40_2, "3": rec_40_3}
+        }
+        load_many_select_features.return_value = expected_result
+
+        resp = self.client.get('/api/v1/low-level/select?recording_ids={}&features={}'.format(recordings, features))
+        self.assertEqual(200, resp.status_code)
+        self.assertEqual(expected_result, resp.json)
+
+        recordings = [("c5f4909e-1d7b-4f15-a6f6-1af376bc01c9", 0),
+                      ("405a5ff4-7ee2-436b-95c1-90ce8a83b359", 2),
+                      ("405a5ff4-7ee2-436b-95c1-90ce8a83b359", 3)]
+        features = [("llj.data->'lowlevel'->'average_loudness'", "lowlevel.average_loudness", None),
+                    ("llj.data->'rhythm'->'onset_rate'", "rhythm.onset_rate", None),
+                    ("llj.data->'metadata'->'version'", "metadata.version", {}),
+                    ("llj.data->'metadata'->'audio_properties'", "metadata.audio_properties", {})]
+        load_many_select_features.assert_called_with(recordings, features)
+
+        # upper-case MBID
+        recordings = "c5f4909e-1d7b-4f15-a6f6-1AF376BC01C9"
+        features = "lowlevel.average_loudness"
+        expected_result = {
+            "c5f4909e-1d7b-4f15-a6f6-1af376bc01c9": {"0": rec_c5}
+        }
+        load_many_select_features.return_value = expected_result
+        resp = self.client.get('/api/v1/low-level/select?recording_ids={}&features={}'.format(recordings, features))
+        self.assertEqual(resp.status_code, 200)
+
+        # We expect that we get returned whatever we set the mock to, but the argument
+        # to load_many_high_level is always lower-case regardless of what we pass in
+        self.assertEqual(resp.json, expected_result)
+        recordings = [("c5f4909e-1d7b-4f15-a6f6-1af376bc01c9", 0)]
+        features = [("llj.data->'lowlevel'->'average_loudness'", "lowlevel.average_loudness", None),
+                    ("llj.data->'metadata'->'version'", "metadata.version", {}),
+                    ("llj.data->'metadata'->'audio_properties'", "metadata.audio_properties", {})]
+        load_many_select_features.assert_called_with(recordings, features)
```

Comment:
@alastair this assertion will fail until PR #349 adds the conversion to lowercase in the _parse_bulk_params method

Reply:
Is this because you proactively added a test for this specific case, even though the previous PR hasn't been merged yet? This is interesting. Normally I would say that this specific PR should be standalone, and that you shouldn't write a test that intentionally fails, but you're right that it's a good idea to have this test eventually, and we need to keep track somehow that it's necessary.
One thing we could do here is mark this test function with `@unittest.skip` so that it's not run, and make a note that it has to be un-skipped when the other is merged.

Reply:
Yes I added it proactively because once we merge #349 we will need it anyways, so my thought was that this would be easier than adding a new PR just for the test once this one and #349 are already merged? I will use `@unittest.skip` and add a comment so we can keep track of it for now.